src/m365/planner/commands/roster/roster-member-remove.spec.ts

```diff
@@ -0,0 +1,303 @@
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.delete,
+      request.get,
+      Cli.prompt
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.ROSTER_MEMBER_REMOVE);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if userId is not a valid guid.', async () => {
+    const actual = await command.validate({
+      options: {
+        rosterId: validRosterId,
+        userId: 'Invalid GUID'
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if required options specified (id)', async () => {
+    const actual = await command.validate({ options: { rosterId: validRosterId, userId: validUserId } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation if required options specified (name)', async () => {
+    const actual = await command.validate({ options: { rosterId: validRosterId, userName: validUserName } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('prompts before removing the specified roster member when confirm option not passed', async () => {
+    await command.action(logger, {
+      options: {
+        rosterId: validRosterId,
+        userid: validUserId
+      }
+    });
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('prompts before removing the last roster member when confirm option not passed', async () => {
+    let secondPromptOptions: any;
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async (options: any) => {
+      if (options.message === `Are you sure you want to remove member '${validUserId}'?`) {
+        return (
+          { continue: true }
+        );
+      }
+      else {
+        secondPromptOptions = options;
+        return (
+          { continue: false }
+        );
+      }
+    });
+
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/beta/planner/rosters/${validRosterId}/members?$select=Id`) {
+        return (singleRosterMemberResponse);
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        rosterId: validRosterId,
+        userId: validUserId
+      }
+    });
+
+    let promptIssued = false;
+
+    if (secondPromptOptions && secondPromptOptions.type === 'confirm' && secondPromptOptions.message === `Are you sure you want to remove the last member from the roster '${validRosterId}'? When the last user is removed, the Roster and all its contents will be deleted within 30 days`) {
```

Comment:
If I'm not mistaken, we can remove the message check here since we already check on the message in the stub. So if the message doesn't match there, `secondPromptOptions` would be `undefined` and thus the test will fail.
Other than that, nice test! 👍